### PR TITLE
T4: tighten test quality — Theory split + capacity-path coverage

### LINKED ISSUE #117: [Maintenance] testing: Audit test quality
## Scope

Audit the existing test suite for quality issues that pass CI but reduce real signal: tests that do not assert, weak assertions where stronger ones exist (Assert.True(x != null) -> Assert.NotNull), Assert.Equal(null, ...) instead of Assert.Null, test naming pattern violations (MethodUnderTest_when_condition_expected_result), shared mutable state across test runs, tests of implementation details rather than behavior, hardcoded magic numbers (e.g. 9999999L -> TimeSpan.TicksPerSecond - 1), [Theory] inline data that should be MemberData, missing boundary-case tests for paths the source explicitly handles (MinValue/MaxValue clamps, null-input short-circuits). Run as Phase 4.5 of the code-review skill.

## Acceptance criteria

Each finding categorized as must-fix / should-fix / nit. Must-fix and should-fix items either filed as separate sub-issues or addressed inline. nit items reviewed and either addressed, deferred, or closed as intentional style preference.

## Notes

- Tier-1 foundation task — part of fleet initiative **T4: Audit test quality**.
- Repo-specific scope; assess case by case.
- Board: https://github.com/users/Chris-Wolfgang/projects/6

### PR DESCRIPTION
**Pull request by Chris-Wolfgang**
## Summary

Closes #117 (Audit test quality — T4 canonical initiative).

### 1. Combined-scenario Facts split into Theory + per-type cases

`IsEmpty_when_source_is_HashSet_returns_correct_result` and
`IsNotEmpty_when_source_is_LinkedList_returns_correct_result` each
exercised two scenarios in a single `[Fact]` — refactored into
`[Theory]` methods backed by `TheoryData` covering `List<T>`,
`HashSet<T>`, `LinkedList<T>`, and `Collection<T>` empty + non-empty
pairs (8 cases per Theory). Failures now report per-scenario instead
of hiding behind a combined assert.

### 2. Capacity-pre-allocation path coverage

`AddRange`'s `List<T> + ICollection<T>` capacity optimisation
(`ICollectionExtensions.cs` lines 86-89) had no test coverage. Added
two `[Fact]`s:

- `AddRange_when_source_is_List_and_items_is_ICollection_pre_allocates_capacity`
  proves Capacity is set to the exact final count rather than
  overshooting via List's amortised doubling.
- `AddRange_when_source_List_already_has_sufficient_capacity_does_not_shrink_it`
  proves the `Math.Max(current, needed)` guard preserves
  over-allocations.

### 3. LinkedListWith helper

Added a private `LinkedListWith<T>(params)` helper since
`LinkedList<T>`'s collection-initialiser support varies by TFM.

## Test count

23 → 39 (16 Theory cases + 2 capacity Facts).

## T4 audit findings deferred / not actioned

- Raw `Assert.True(result)` / `Assert.False(result)` on bool-returning
  methods: not changed. xunit has no stronger matcher for a raw bool
  return; the assertion is already idiomatic.
- "Not thread-safe" doc claim is not exercised by a concurrency test:
  defensive doc only; not worth a flaky concurrency test in a public
  extension method whose synchronisation responsibility is documented
  as the caller's.

### CODE REVIEW

File: tests/Wolfgang.Extensions.ICollection.Tests.Unit/ICollectionExtensionTests.cs [hunk 1]
```diff
@@ -376,16 +392,89 @@ public void IsNotEmpty_when_source_has_items_returns_true()
...
+        // relying on List's amortised doubling.
+        var source = new List<int>(capacity: 1) { 0 };
+        var items = new List<int> { 1, 2, 3, 4, 5, 6, 7, 8, 9 };
 
-        // Act & Assert
-        Assert.False(empty.IsNotEmpty());
-        Assert.True(nonEmpty.IsNotEmpty());
+        // Act
+        source.AddRange(items);
+
```
[the remark this answers is not in the document]
Comment (Chris-Wolfgang) [reply tests/Wolfgang.Extensions.ICollection.Tests.Unit/ICollectionExtensionTests.cs:438]: Addressed on PR #127 (the proper stacked version of this T4 work). Same capacity-binding fix landed in commit d1160e8 on feature/t4-test-quality-audit-v2 — receiver is typed as ICollection<int> so the extension wins overload resolution. Closing this PR as superseded.

File: tests/Wolfgang.Extensions.ICollection.Tests.Unit/ICollectionExtensionTests.cs [hunk 2]
```diff
@@ -376,16 +392,89 @@ public void IsNotEmpty_when_source_has_items_returns_true()
...
+    [Fact]
+    public void AddRange_when_source_List_already_has_sufficient_capacity_does_not_shrink_it()
+    {
+        // Arrange — list with deliberately oversized capacity.
+        var source = new List<int>(capacity: 100) { 0 };
+        var items = new List<int> { 1, 2, 3 };
+
+        // Act
+        source.AddRange(items);
+
```
[the remark this answers is not in the document]
Comment (Chris-Wolfgang) [reply tests/Wolfgang.Extensions.ICollection.Tests.Unit/ICollectionExtensionTests.cs:458]: Addressed on PR #127 (the proper stacked version) — commit d1160e8 types the receiver as ICollection<int> for the Math.Max(current, needed) test too.

File: tests/Wolfgang.Extensions.ICollection.Tests.Unit/ICollectionExtensionTests.cs [hunk 3]
```diff
@@ -315,16 +317,30 @@ public void IsEmpty_when_source_has_items_returns_false()
 
 
 
-    [Fact]
-    public void IsEmpty_when_source_is_HashSet_returns_correct_result()
+    public static TheoryData<ICollection<int>, bool> IsEmpty_collection_type_cases =>
```
[the remark this answers is not in the document]
Comment (Chris-Wolfgang) [reply tests/Wolfgang.Extensions.ICollection.Tests.Unit/ICollectionExtensionTests.cs:320]: Fixed on PR #127 (commit d549419) — renamed to IsEmptyCollectionTypeCases per the .editorconfig non_field_members_should_be_pascal_case rule.

File: tests/Wolfgang.Extensions.ICollection.Tests.Unit/ICollectionExtensionTests.cs [hunk 4]
```diff
@@ -376,16 +392,89 @@ public void IsNotEmpty_when_source_has_items_returns_true()
 
 
 
+    public static TheoryData<ICollection<int>, bool> IsNotEmpty_collection_type_cases =>
```
[the remark this answers is not in the document]
Comment (Chris-Wolfgang) [reply tests/Wolfgang.Extensions.ICollection.Tests.Unit/ICollectionExtensionTests.cs:395]: Fixed on PR #127 (commit d549419) — renamed to IsNotEmptyCollectionTypeCases per the same PascalCase rule.

**Comment by Chris-Wolfgang** [comment]
Superseded by PR #127 (the stacked T4 work that lands as part of the v0.3.0 release pilot). All 4 review findings from this PR were applied to feature/t4-test-quality-audit-v2 — capacity-test binding fix in commit d1160e8, PascalCase rename in commit d549419.
